resolwe_bio/processes/alignment/subread.yml

```diff
@@ -0,0 +1,169 @@
+  input:
+    - name: genome
+      label: Reference genome
+      type: data:genome:fasta
+    - name: reads
+      label: Reads
+      type: data:reads:fastq
+    - name: options
+      label: Options
+      group:
+        - name: indel
+          label: Number of INDEL bases
+          type: basic:integer
+          required: false
+          default: 5
+          description: >
+            Specify the number of INDEL bases allowed in the mapping.
+        - name: consensus
+          label: Consensus threshold
+          type: basic:integer
+          required: false
+          default: 3
+          description: >
+            Specify the consensus threshold, which is the minimal number of consensus subreads required for reporting a hit.
+        - name: mis_matched_bp
+          label: Max number of mis-matched bases
+          type: basic:integer
+          required: false
+          default: 3
+          description: >
+            Specify the maximum number of mis-matched bases allowed in the alignment.
+        - name: cpu_number
+          label: Number of threads/CPUs
+          type: basic:integer
+          required: false
+          default: 1
+          description: >
+            Specify the number of threads/CPUs used for mapping
+        - name: unique_reads
+          label: Output uniquely mapped reads
+          type: basic:boolean
+          required: false
+          description: >
+            Reads that were found to have more than one best mapping location will not be reported.
+    - name: PE_options
+      label: Paired end alignment options
+      group:
+        - name: reads_orientation
+          label: reads orientation
+          type: basic:string
+          required: false
+          description: >
+            Specify the orientation of the two reads from the same pair.
+          default: "fr"
+          choices:
+            - label: ff
+              value: ff
+            - label: fr
+              value: fr
+            - label: rf
+              value: rf
+        - name: consensus_subreads
+          label: Minimum number of consensus subreads
+          type: basic:integer
+          required: false
+          default: 1
+          description: >
+            Specify the minimum number of consensus subreads both reads from the sam pair must have.
+  output:
+    - name: bam
+      label: Alignment file
+      type: basic:file
+      description: Position sorted alignment
+    - name: bai
+      label: Index BAI
+      type: basic:file
+    - name: unmapped
+      label: Unmapped reads
+      type: basic:file
+      required: false
+    - name: stats
+      label: Statistics
+      type: basic:file
+  run:
+    runtime: polyglot
+    bash: |
+      INDEX="{{genome.index_subread.dir}}/genome_index"
+
+      if [ {{ genome.index_subread.dir }} ]; then
+        echo "Genome index found."
+      else
+        echo "Index not found, provide index file."
```

Comment:
It would be better to use `re-error "..."` instead of `echo`. If there is no index for subread in the genome data_object, the processor should fail already at this stage.